tools/common_compiler_flags.py

```diff
@@ -64,6 +69,22 @@ def generate(env):
             env.Append(LINKFLAGS=["/OPT:REF"])
         elif env["optimize"] == "debug" or env["optimize"] == "none":
             env.Append(CCFLAGS=["/Od"])
+
+        if env["lto"] == "thin":
+            if not env["use_llvm"]:
+                print("ThinLTO is only compatible with LLVM, use `use_llvm=yes` or `lto=full`.")
+                sys.exit(255)
+
+            env.Append(CCFLAGS=["-flto=thin"])
+            env.Append(LINKFLAGS=["-flto=thin"])
+        elif env["lto"] == "full":
+            if env["use_llvm"]:
+                env.Append(CCFLAGS=["-flto"])
+                env.Append(LINKFLAGS=["-flto"])
+            else:
+                env.AppendUnique(CCFLAGS=["/GL"])
+                env.AppendUnique(ARFLAGS=["/LTCG"])
+                env.AppendUnique(LINKFLAGS=["/LTCG"])
```

Comment:
This is what I thought as well, but I'm mirroring what Godot does: https://github.com/godotengine/godot/blob/master/platform/windows/detect.py#L620
(note it's the MSVC branch this is in)

If you think this is not a real scenario, maybe this is worth a PR into the main repository as well? Let me know if I should still change it according to your suggestion.

Reply:
Mmmhhh.. the `-` vs `/` flags upstream is suspicious, but I think more importantly there's a discrepancy between godot-cpp and godot where we use `use_clang_cl` instead of `use_llvm` in godot-cpp for the llvm msvc driver.

So currently, that scenario can never happen in godot-cpp, but I think we should unify the `use_clang_cl` flag with `use_llvm` and, like upstream godot does(?), use the llvm MSVC backend unless `use_mingw` is also specified.

But we should probably double check this with someone more into the windows part of the build system.

Reply:
I've opened #1602 to discuss that change specifically.

Reply:
With #1602 merged, this issue should be resolved.